### PR TITLE
make the etching fluid for empty PCBs its own method for customizability and better addon compat

### PR DESCRIPTION
**Pull request by RemagOfficial**
idk why the commits from my old attempt at fixing custom pcbs are included but i dont think i can remove them, the only stuff that is different is the stuff from the latest commit

### CODE REVIEW

File: src/main/java/me/desht/pneumaticcraft/common/block/entity/EtchingTankBlockEntity.java [hunk 1]
```diff
@@ -328,8 +328,8 @@ private class EtchingFluidTank extends SmartSyncTank {
         }
 
         @Override
-        public boolean isFluidValid(FluidStack stack) {
-            return stack.getFluid().is(PneumaticCraftTags.Fluids.ETCHING_ACID);
+        public boolean isFluidValid(FluidStack fluidStack) {
```
Comment (desht) [inline src/main/java/me/desht/pneumaticcraft/common/block/entity/EtchingTankBlockEntity.java:331]: Why this change?  What's wrong with checking for the fluid tag here?

**Review by desht — CHANGES_REQUESTED** [review]

File: src/main/java/me/desht/pneumaticcraft/common/block/entity/EtchingTankBlockEntity.java (hunk 1 above)
Comment (RemagOfficial) [reply src/main/java/me/desht/pneumaticcraft/common/block/entity/EtchingTankBlockEntity.java:331]: if I check the tag it means custom empty PCBs all have to use the same fluids for etching, for example you might want a custom PCB that needs it's own etching fluid to etch rather than using the vanilla pneumaticcraft one
